src/lib/components/Instances/InstanceTable.svelte

```diff
@@ -47,9 +49,17 @@
 				.map((d, i) => (d ? instances[i] : null))
 				.filter((d) => d !== null) as Tables<'instances'>[];
 		}
-		generateInstances(prompt, passedInstances, count, instruction).then((r) => {
-			instances = [...instances, ...r];
-			generatingInstances = false;
+
+		generateInstances(prompt, passedInstances, count, instruction).then((newInstances) => {
+			getPredictions(prompt, newInstances).then((newPredictions) => {
+				const localPreds = { ...predictions };
+				newPredictions?.forEach(
+					(d, i) => (localPreds[newInstances[i].id] = new Promise((resolve) => resolve(d)))
+				);
+				predictions = localPreds;
+				instances = [...instances, ...newInstances];
+				generatingInstances = false;
+			});
```

Comment:
What's the advantage of `getPredictions` over `getPrediction`?

With the latter you could just do:
```
generateInstances(prompt, passedInstances, count, instruction).then((newInstances) => {
  const tmpPreds = {...predictions};
  for (const newInstance in newInstances) {
    localPreds[newInstance.id] = getPrediction(prompt, newInstance);
  }
  predictions = localPreds;
  instances = [...instances, ...newInstances];
 	generatingInstances = false;
});
```

Which I think would look cleaner.

Reply:
A bunch of get requests could be a limitation but I think in this case since we're only generating 5-10 at a time it should be fine and would be cleaner!